feat(namemc): init

Nyatalieeee opened the pull request:
<!-- Replace "Website" with a markdown link to the website that you have themed. -->

## 🎉 Theme for [NameMC](https://namemc.com/) 🎉

NameMC is a website for viewing Minecraft skins, servers, usernames, players, etc.

![skins](https://github.com/user-attachments/assets/0f8ac512-e98a-42a6-a63d-f7ed9a276ae8)
![names](https://github.com/user-attachments/assets/a35c60b8-72ed-4985-853e-557e543eeb01)
![servers](https://github.com/user-attachments/assets/2723eb0d-6138-4bf4-943e-f182a8cfd0cf)
![capes](https://github.com/user-attachments/assets/593f136f-fd51-45e8-89ea-210387fa80a0)
![search](https://github.com/user-attachments/assets/7c712283-84c5-47e7-89e1-475951307c37)
![profile](https://github.com/user-attachments/assets/3f3316a2-b8fb-4a1e-9ca8-92d6be7e80a4)

## 💬 Additional Comments 💬

<!--
Include any difficulties you had theming this port, or any general comments that would be useful for the reviewer to know.
Feel free to leave this section empty if you don't have anything more to say.
-->

## 🗒 Checklist 🗒

- [x] I have read and followed Catppuccin's [submission guidelines](https://github.com/catppuccin/userstyles/blob/main/docs/userstyle-creation.md).
- [x] I have made a new directory underneath `/styles/<name-of-website>` containing the contents of the [`/template`](https://github.com/catppuccin/userstyles/blob/main/template/) directory.
  - [x] I have ensured that the new directory is in **lower-kebab-case**.
  - [x] I have followed the template and kept the preprocessor as [LESS](https://lesscss.org/#overview).
- [x] I have made sure to update the
      [`userstyles.yml`](https://github.com/catppuccin/userstyles/blob/main/scripts/userstyles.yml)
      file with information about the new userstyle.
- [x] I have included the following files:
  - [x] `catppuccin.user.css` - all the CSS for the userstyle, based on the
        template.
  - [x] `preview.webp` - composite image of all four individual flavor screenshots (taken with the default accent color of mauve) stitched together, generated via [Catwalk](https://github.com/catppuccin/catwalk).

isabelroses commented on styles/namemc/catppuccin.user.css:
```diff
@@ -0,0 +1,765 @@
+    --bs-success-bg-subtle: @crust;
+    --bs-info-bg-subtle: @crust;
+    --bs-warning-bg-subtle: @surface0;
+    --bs-danger-bg-subtle: @base;
+    --bs-light-bg-subtle: @surface0;
+    --bs-dark-bg-subtle: @mantle;
+    --bs-primary-border-subtle: @mantle;
+    --bs-secondary-border-subtle: @surface1;
+    --bs-success-border-subtle: @accent-color;
+    --bs-info-border-subtle: @teal;
+    --bs-warning-border-subtle: @peach;
+    --bs-danger-border-subtle: @red;
+    --bs-light-border-subtle: @surface1;
+    --bs-dark-border-subtle: @surface0;
+    --bs-heading-color: inherit;
+    --bs-link-color: @accent-color;
+    --bs-link-hover-color: @text;
+    --bs-link-color-rgb: @accent-color;
+    --bs-link-hover-color-rgb: @text;
+    --bs-code-color: @accent-color;
+    --bs-border-color: @surface1;
+    --bs-border-color-translucent: @surface1;
+    --bs-form-valid-color: @green;
+    --bs-form-valid-border-color: @green;
+    --bs-form-invalid-color: @red;
+    --bs-form-invalid-border-color: @red;
+
+    .mc-black {
+      color: @crust !important;
+    }
+    .mc-dark-gray {
+      color: @overlay0 !important;
+    }
+    .mc-gray,
+    .mc-reset {
+      color: @subtext0 !important;
+    }
+    .mc-white {
+      color: @text !important;
+    }
```
you can deduplicate this by moving it to the `#catppuccin()` mixin with no additional selectors

isabelroses commented on styles/namemc/catppuccin.user.css:
```diff
@@ -0,0 +1,765 @@
+    }
+    .btn-outline-danger {
+      --bs-btn-color: @red;
+      --bs-btn-border-color: @red;
+      --bs-btn-hover-color: @text;
+      --bs-btn-hover-bg: @red;
+      --bs-btn-hover-border-clor: @red;
+      --bs-btn-focus-shadow-rgb: @red;
+      --bs-btn-active-color: @text;
+      --bs-btn-active-bg: @red;
+      --bs-btn-active-border-color: @red;
+      --bs-btn-disabled-color: @red;
+      --bs-btn-disabled-border-color: @red;
+    }
+    .btn-outline-light {
+      --bs-btn-color: @text;
+      --bs-btn-border-color: @text;
+      --bs-btn-hover-color: @crust;
+      --bs-btn-hover-bg: @text;
+      --bs-btn-hover-border-color: @text;
+      --bs-btn-focus-shadow-rgb: @text;
+      --bs-btn-active-color: @crust;
+      --bs-btn-active-bg: @text;
+      --bs-btn-active-border-color: @text;
+      --bs-btn-disabled-color: @text;
+      --bs-btn-disabled-border-color: @text;
+    }
+    .btn-outline-dark {
+      --bs-btn-color: @surface0;
+      --bs-btn-border-color: @surface0;
+      --bs-btn-hover-color: @text;
+      --bs-btn-hover-bg: @surface0;
+      --bs-btn-hover-border-color: @surface0;
+      --bs-btn-focus-shadow-rgb: @surface0;
+      --bs-btn-active-color: @text;
+      --bs-btn-active-bg: @surface0;
+      --bs-btn-active-border-color: @surface0;
+      --bs-btn-disabled-color: @surface0;
+      --bs-btn-disabled-border-color: @surface0;
+    }
```
this seems like a lot of repetition, it might be worth abstracting it into a mixin

uncenter commented on scripts/userstyles.yml:
```diff
@@ -622,6 +622,14 @@ userstyles:
     current-maintainers: [*gitmuslim]
+  namemc:
+    name: NameMC
+    categories: [game, entertainment]
```
```suggestion
    categories: [game_development, analytics, discussion_forum]
```

This feel "more right" to me. It isn't a game itself, nor does it provide entertaniment per se. Game *development* seems more apt here, analytics since it tracks accounts in the game, and discussion forum since people can post/upload/share skins and stuff. I'm also now realizing we may want to adjust the categories of Planet Minecraft as well :)

Nyatalieeee commented:
i definitely agree with the changes to userstyles.yml, those categories seem more right to me as well tbh

though i'm not sure i understand what those changes to the css would involve, could you clarify on that? sorry

isabelroses commented:
> though i'm not sure i understand what those changes to the css would involve, could you clarify on that? sorry

For https://github.com/catppuccin/userstyles/pull/1207#discussion_r1720202337 see [this commit from pr #1199](https://github.com/catppuccin/userstyles/pull/1199/commits/df82f929edf2e90d0fea792ea009fd9795f425f1). And for the other point 

For https://github.com/catppuccin/userstyles/pull/1207#discussion_r1720203804 see
- https://github.com/catppuccin/userstyles/blob/b5e8b8938ccbd2581df93b6bc38a8fad477bd0e7/styles/vikunja/catppuccin.user.css#L48
- https://github.com/catppuccin/userstyles/blob/b3528f75a436892b374528e41bc7123868f8f8e1/styles/github/catppuccin.user.css#L61

Nyatalieeee commented:
does this look good? i converted the buttons to mixins and moved the other vars to the `#catppuccin()` mixin, though i struggled to understand how mixins work at first so i'm not sure if it's exactly what you imagined

isabelroses reviewed (changes requested):
thanks for fixing my prior comments. Still a bit more work before we can merge this

Nyatalieeee commented:
sorry i've been inactive the past week, i've been very busy irl lol

anyway does this look better?

uncenter commented on styles/namemc/catppuccin.user.css:
```diff
@@ -0,0 +1,582 @@
+      #bsButtonOutline(@surface0, @text);
+    }
+
+    color-scheme: if(@lookup = latte, light, dark);
+
+    ::selection {
+      background-color: fade(@accent-color, 30%);
+    }
+
+    input,
+    textarea {
+      &::placeholder {
+        color: @subtext0 !important;
+      }
+    }
```
This should be the first section of stuff, right beneath the variables (line 54).

uncenter commented on styles/namemc/catppuccin.user.css:
```diff
@@ -0,0 +1,582 @@
+
+          button.btn.btn-primary {
+            background-color: @accent-color;
+            transition: all linear 0.15s;
```
Why is this set?

uncenter commented on styles/namemc/catppuccin.user.css:
```diff
@@ -0,0 +1,582 @@
+      background-color: @mantle !important;
+    }
+
+    table.server-table {
+      > tbody > tr > td > a {
+        color: @accent-color !important;
+      }
+    }
```
Are you sure there isn't a better selector for this? No class, ID, inline style, attribute, etc? This is unclear what it themes.

uncenter reviewed (commented):
A lot of borders seem to have gone missing (around inputs, buttons, and bottom borders below text/headings), and that date in the center has lost its accent flavored-ness.

| Before | After |
| --- | --- |
| ![image](https://github.com/user-attachments/assets/2e491bea-1d6a-48ab-ae6f-fc4d2383b2e9) | ![image](https://github.com/user-attachments/assets/e56692b9-5452-427c-8f7d-4050b663e17e) |

uncenter commented on styles/namemc/catppuccin.user.css:
```diff
@@ -0,0 +1,582 @@
+      --bs-btn-disabled-color: @col;
+
+      --bs-btn-bg: @bgCol;
+      --bs-btn-hover-bg: @bgCol;
+      --bs-btn-active-bg: @bgCol;
```
The hover/active color should be different from the first color. `darken(@accent-color, 5%)` would work.

uncenter commented on styles/namemc/catppuccin.user.css:
```diff
@@ -0,0 +1,582 @@
+
+    table {
+      tbody tr {
+        td.text-nowrap.text-ellipsis a {
+          color: @accent-color !important;
+        }
```
Why override the link styling from the variables above here?

uncenter reviewed (commented):
All of the `*-rbg` variables don't work/correctly theme anything. You need to use https://github.com/catppuccin/userstyles/blob/main/docs/tips-and-tricks.md#how-do-i-set-a-variable-to-rgb-values. For example, this breaks footer links due to `rgba(#cba6f7, 1);` not being valid.

uncenter commented on styles/namemc/catppuccin.user.css:
```diff
@@ -0,0 +1,582 @@
+    }
+
+    /* skin display background */
+    .checkered {
+      background-image: none !important;
+      background-color: @mantle !important;
+    }
```
We can make this look way better:

```css
    .checkered {
        @dark: fade(@surface0, 40%);
        @light: fade(@surface1, 40%);
        @svg: escape('<svg width="64" height="64" xmlns="http://www.w3.org/2000/svg"><path fill="@{dark}" d="M0 0h32v32H0z"/><path fill="@{light}" d="M32 0h32v32H32z"/><path fill="@{light}" d="M0 32h32v32H0z"/><path fill="@{dark}" d="M32 32h32v32H32z"/></svg>');
      background-image: url("data:image/svg+xml,@{svg}");
    }
```

uncenter commented on styles/namemc/catppuccin.user.css:
```diff
@@ -0,0 +1,582 @@
+        }
+      }
+
+      *.form-control,
+      *.form-select {
+        background-color: @mantle !important;
+        border-color: @surface0 !important;
+        box-shadow: none !important;
+        color: @subtext1 !important;
+
+        &:focus {
+          border-color: @accent-color !important;
+        }
```
These shouldn't be scoped just for this page, the form-* elements are used all over.

uncenter replied:
Also for the &:focus you need `box-shadow: 0 0 0 .25rem fade(@accent-color, 25%)`.

uncenter reviewed (commented):
`--bs-dropdown-header-color` unthemed ("Profile" text in your profile picture dropdown).

uncenter commented on styles/namemc/catppuccin.user.css:
```diff
@@ -0,0 +1,582 @@
+      --bs-btn-active-border-color: @col;
+      --bs-btn-disabled-border-color: @col;
+
+      --bs-btn-hover-color: @text;
+      --bs-btn-active-color: @text;
```
You forgot to change these to use `@hover` instead of `@text`.

uncenter commented:
That's all I have for now. Please incorporate fixes to the above highlighted problems in a followup PR.
